docs/arch/adr-3-store-status.md

```diff
@@ -0,0 +1,123 @@
+
+That worked until now, when there were only had 4 statuses:
+
+* draft
+* live
+* scheduled
+* unpublished changes
+
+Implementing the 2i workflow will need the addition of least three more statuses:
+
+* submitted for 2i
+* in review
+* 2i approved
+
+This will make the code to determine the status from the surrounding information very complicated.
+
+### What status information is needed?
+
+As well as knowing what part of the 2i workflow a step-by-step is in, the step-by-step also needs to know if it has been previously published. This is so that the correct options are given to users to discard their changes. For example, if a step-by-step has previously been published, it can’t be deleted.
+
+
+### 2i workflows
+
+There are two "happy path" 2i workflows that a step-by-step can follow
+
+### A new step-by-step is created
+
+|Action                       |Publication State|Workflow State      |
+|-----------------------------|:---------------:|-------------------:|
+|Create new step-by-step      |Draft            |N/A                 |				
+|Makes changes to step-by-step|Draft            |N/A                 |				
+|Submit for 2i                |Draft            |Submitted for 2i    |
+|Claim Review                 |Draft            |In Review           |
+|2i Approved                  |Draft            |2i Approved         |
+|2i Requests change           |Draft            |N/A                 |
+|Publish                      |Published        |N/A                 |
+|Schedule                     |Draft            |Scheduled           |
+|Unschedule                   |Draft            |2i Approved         |
+|Delete                       |Draft            |N/A                 |
+
+### A published step-by-step is updated
+
+|Action                       |Publication State  |Workflow State      |
+|-----------------------------|:-----------------:|-------------------:|
+|Update published step-by-step|Published          |N/A                 |
+|Make changes to step-by-step |Unpublished changes|N/A                 |
+|Submit for 2i                |Unpublished changes|Submitted for 2i    |
+|Claim Review                 |Unpublished changes|In Review           |
+|2i Approved                  |Unpublished changes|2i Approved         |
+|2i Requests changes          |Unpublished changes|N/A                 |
+|Publish                      |Published          |N/A                 |
+|Schedule                     |Unpublished changes|Scheduled           |
+|Unschedule                   |Unpublished changes|2i Approved         |
+|Discard Draft                |Published          |N/A                 |
+|Unpublish                    |Draft              |N/A                 |
+
+
+### Status assumptions
+
+- Can only publish if workflow state is 2i Approved
+- Can only schedule if workflow state is 2i Approved
+- Can only Claim Review if workflow state is Submitted for 2i and the Claimer is not the same user as the one that submitted for 2i
```

Comment:
Does this allow the provision for `skip_review` similar to Publisher?  In that scenario, you need a specific permission.  It's particularly useful when testing things on (eg) integration.

Reply:
No. Not at the moment. But I see your point about making it easier to test.